### PR TITLE
[risk=no][DB-597]clickable top-result-chart & chart restructure begin

### PR DESCRIPTION
**Pull request by mhigbyflowers**
This PR Contains the new Chart component system + the feature where the top results can be clicked and corresponding rows will open up.

### CODE REVIEW

File: public-ui/src/app/data-browser/db-table/db-table.component.html [hunk 1]
```diff
@@ -0,0 +1,267 @@
+<div class="results-grid" *ngIf="items && items.length > 0">
```
Comment (mhigbyflowers) [inline public-ui/src/app/data-browser/db-table/db-table.component.html:1]: Added a db-table component to simplify the ehr component

File: public-ui/src/app/data-browser/charts/chart-base/chart-base.component.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import { Component, Injector, Input, } from '@angular/core';
```
Comment (mhigbyflowers) [inline public-ui/src/app/data-browser/charts/chart-base/chart-base.component.ts:1]: This is now the base of all of the chart components. It holds all of the globals. I'm pretty sure there can’t be new Highcharts elements such as "barChartOptions" without being a variable in the base component first

File: public-ui/src/app/data-browser/charts/chart-top-results/chart-top-results.component.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import { Component, EventEmitter, Injector, OnChanges, Output } from '@angular/core';
```
Comment (mhigbyflowers) [inline public-ui/src/app/data-browser/charts/chart-top-results/chart-top-results.component.ts:1]: Chart-top-results extends chart-base 
also new naming convention for chart components

**Review by SrushtiGangireddy — APPROVED** [review]
Some extra whitespaces in there. Otherwise looks good to me.
Thanks.

**Pull request merged**